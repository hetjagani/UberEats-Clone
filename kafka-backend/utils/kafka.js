const sendData = (producer, data, isError) => {
  producer.send(
    [{ topic: 'response', messages: JSON.stringify({ data, isError }), partitions: 1 }],
    (err, d) => {
      if (err) {
        console.error(err);
        throw Error('error sending data to kafka');
      } else {
        console.log('SENT DATA:', d);
      }
    },
  );
};

module.exports = {
  sendData,
};
