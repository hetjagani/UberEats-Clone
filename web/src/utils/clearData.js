export const clearData = () => {
  document.cookie.split(';').forEach(function (c) {
    document.cookie = c
      .replace(/^ +/, '')
      .replace(/=.*/, '=;expires=' + new Date().toUTCString() + ';path=/');

    window.localStorage.removeItem('persist:root');
  });
};
