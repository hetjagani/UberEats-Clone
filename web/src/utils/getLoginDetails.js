import jwt from 'jsonwebtoken';
import { getCookie } from 'react-use-cookie';

const getLoginDetails = () => {
  const token = getCookie('auth');
  if (token) {
    const tokendata = jwt.decode(token);
    return tokendata;
  } else {
    return {};
  }
};

export default getLoginDetails;
