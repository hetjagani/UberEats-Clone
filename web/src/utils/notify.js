import { toast } from 'react-toastify';

const notify = ({ type, description }) => {
  toast[type](`${type}: ${description}`, {
    position: 'top-center',
    autoClose: 5000,
    hideProgressBar: false,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    progress: undefined,
  });
};

export default notify;
