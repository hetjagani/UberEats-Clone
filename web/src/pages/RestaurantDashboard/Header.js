import * as React from 'react';
import { useStyletron } from 'baseui';
import { Layer } from 'baseui/layer';
import { AppNavBar, setItemActive } from 'baseui/app-nav-bar';
import { useSelector } from 'react-redux';
import { RiEdit2Line, RiLogoutBoxRLine } from 'react-icons/ri';
import { CgList } from 'react-icons/cg';
import { AiOutlineUser } from 'react-icons/ai';
import { GrHome } from 'react-icons/gr';
import { useHistory } from 'react-router';
import { clearData } from '../../utils/clearData';

const Header = () => {
  const [css, theme] = useStyletron();

  const loginRestaurant = useSelector((state) => {
    return state.restaurants.loginRestaurant;
  });

  const [userItems, setUserItems] = React.useState([
    { icon: GrHome, label: 'Home' },
    { icon: AiOutlineUser, label: 'Customers' },
    { icon: CgList, label: 'Orders' },
    { icon: RiEdit2Line, label: 'Edit Profile' },
    { icon: RiLogoutBoxRLine, label: 'Logout' },
  ]);

  const imgURL =
    loginRestaurant.media && loginRestaurant.media.length > 0 && loginRestaurant.media[0].url;

  const history = useHistory();

  const handelNavItem = (item) => {
    switch (item.label) {
      case 'Home':
        history.push('/dashboard');
        break;
      case 'Customers':
        history.push('/customers');
        break;
      case 'Edit Profile':
        history.push('/details/update');
        break;
      case 'Orders':
        history.push('/orders/restaurants');
        break;
      case 'Logout':
        clearData();
        window.location.href = '/auth/login';
        break;
    }
  };

  return (
    <div>
      <Layer>
        <div
          className={css({
            boxSizing: 'border-box',
            width: '100vw',
            position: 'fixed',
            top: '0',
            left: '0',
          })}
        >
          <AppNavBar
            onMainItemSelect={() => alert(1)}
            title={loginRestaurant.name}
            userItems={userItems}
            onUserItemSelect={handelNavItem}
            username={loginRestaurant.name}
            userImgUrl={imgURL}
          />
        </div>
      </Layer>
    </div>
  );
};

export default Header;
