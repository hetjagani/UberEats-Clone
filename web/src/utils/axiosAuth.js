import axios from 'axios';
import { getCookie, setCookie } from 'react-use-cookie';
import notify from './notify';

function createAxiosAuthMiddleware() {
  return ({ getState }) =>
    (next) =>
    (action) => {
      const authToken = getCookie('auth');

      if (authToken) {
        axios.defaults.headers.common['Authorization'] = authToken;
      }

      axios.defaults.baseURL = window.BACKEND_API_URL;
      axios.defaults.withCredentials = false;
      axios.interceptors.response.use(
        (res) => res,
        (err) => {
          if (err.response.status === 401) {
            setCookie('auth', null);
            window.location = '/auth/login';
            notify({ type: 'error', description: 'Unauthorized. Please Login.' });
          }
          return Promise.reject(err);
        },
      );
      return next(action);
    };
}

export default createAxiosAuthMiddleware;
