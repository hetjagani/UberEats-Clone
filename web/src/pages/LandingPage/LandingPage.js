import { useStyletron } from 'baseui';
import React, { useEffect, useState } from 'react';
import { H1 } from 'baseui/typography';
import Header from './Header';
import { Drawer, ANCHOR, SIZE } from 'baseui/drawer';
import { Button, SIZE as BSIZE } from 'baseui/button';
import { Input, SIZE as ISIZE, ADJOINED } from 'baseui/input';
import { Link, useHistory } from 'react-router-dom';
import getLoginDetails from '../../utils/getLoginDetails';

const LandingPage = () => {
  const [css, theme] = useStyletron();
  const main = css({
    backgroundImage: 'url("images/uber_landing.webp")',
    backgroundSize: 'cover',
    height: '100vh',
    width: '100vw',
  });

  const container = css({
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    margin: '30px',
    width: '80vw',
    height: '70vh',
  });

  const labelStyle = css({
    color: 'white',
    width: '256px',
    font: theme.typography.LabelMedium,
  });

  const buttonStyle = css({
    marginBottom: '20px',
    marginTop: '20px',
  });

  const history = useHistory();
  useEffect(() => {
    const loginDetails = getLoginDetails();
    if (loginDetails?.role) {
      if (loginDetails.role == 'customer') {
        history.push('/restaurants');
      } else if (loginDetails.role == 'restaurant') {
        history.push('/dashboard');
      }
    }
  }, []);

  const [isOpen, setIsOpen] = useState(false);
  const [email, setEmail] = useState('');

  return (
    <div className={main}>
      <Header setIsOpen={setIsOpen} />
      <div className={container}>
        <H1 className={css({ fontWeight: 'bold' })}>Order food to your door</H1>
        <div className={css({ width: '80%', display: 'flex', justifyContent: 'flex-start' })}>
          <Input
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            placeholder="Enter your email address"
          />
          <Link to={`/auth/registeruser?email=${email}`}>
            <span style={{ marginLeft: '20px', width: 500 }}>
              <Button size={BSIZE.large}>Sign up</Button>
            </span>
          </Link>
        </div>
        <span className={css({ fontWeight: 'bold', margin: '5px' })}>
          Click Here to <Link to={'/auth/registerres'}>add your restaurant</Link>
        </span>
      </div>
      <Drawer
        onClose={() => setIsOpen(false)}
        isOpen={isOpen}
        anchor={ANCHOR.left}
        size={SIZE.auto}
      >
        <div className={buttonStyle}>
          <Link to={'/auth/login'}>
            <Button className={buttonStyle} size={BSIZE.large}>
              <span className={labelStyle}>Sign in</span>
            </Button>
          </Link>
        </div>
        <div className={buttonStyle}>
          <Link to={'/auth/registerres'}>
            <Button className={buttonStyle} size={BSIZE.large}>
              <span className={labelStyle}>Add your restaurant</span>
            </Button>
          </Link>
        </div>
      </Drawer>
    </div>
  );
};

export default LandingPage;
