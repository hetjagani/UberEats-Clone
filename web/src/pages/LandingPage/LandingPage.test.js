import { Drawer } from 'baseui/drawer';
import { H1 } from 'baseui/typography';
import { shallow } from 'enzyme';
import React from 'react';
import { Link } from 'react-router-dom';
import '../../utils/testsetup';
import Header from './Header';
import LandingPage from './LandingPage';

describe('Landing Page Test', () => {
  it('should render Landing page', () => {
    const landingPage = shallow(<LandingPage />);
    expect(landingPage.getElements()).toMatchSnapshot();
  });

  it('should contain a Header and Drawer', () => {
    const landingPage = shallow(<LandingPage />);
    expect(landingPage.find(Header).length).toBe(1);
    expect(landingPage.find(Drawer).length).toBe(1);
  });

  it('should have proper heading', () => {
    const landingPage = shallow(<LandingPage />);
    expect(landingPage.find(H1).text()).toBe('Order food to your door');
  });

  it('should have links to signin and signup', () => {
    const landingPage = shallow(<LandingPage />);
    expect(
      landingPage
        .find(Link)
        .getElements()
        .filter((e) => {
          return e.props.to.includes('/auth/registeruser');
        }).length,
    ).toBeGreaterThan(0);

    expect(
      landingPage
        .find(Link)
        .getElements()
        .filter((e) => {
          return e.props.to.includes('/auth/registerres');
        }).length,
    ).toBeGreaterThan(0);

    expect(
      landingPage
        .find(Link)
        .getElements()
        .filter((e) => {
          return e.props.to.includes('/auth/login');
        }).length,
    ).toBeGreaterThan(0);
  });
});
