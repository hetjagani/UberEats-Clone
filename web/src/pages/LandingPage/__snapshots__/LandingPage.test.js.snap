// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Test should render Landing page 1`] = `
Array [
  <div
    className=""
  >
    <Header
      setIsOpen={[Function]}
    />
    <div
      className=""
    >
      <HeadingXXLarge
        className=""
      >
        Order food to your door
      </HeadingXXLarge>
      <div
        className=""
      >
        <Input
          autoComplete="on"
          autoFocus={false}
          clearable={false}
          disabled={false}
          endEnhancer={null}
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          overrides={Object {}}
          placeholder="Enter your email address"
          required={false}
          size="default"
          startEnhancer={null}
          type="text"
          value=""
        />
        <Link
          to="/auth/registeruser?email="
        >
          <span
            style={
              Object {
                "marginLeft": "20px",
                "width": 500,
              }
            }
          >
            <Button
              size="large"
            >
              Sign up
            </Button>
          </span>
        </Link>
      </div>
      <span
        className=""
      >
        Click Here to 
        <Link
          to="/auth/registerres"
        >
          add your restaurant
        </Link>
      </span>
    </div>
    <Drawer
      anchor="left"
      animate={true}
      autoFocus={true}
      closeable={true}
      isOpen={false}
      onClose={[Function]}
      overrides={Object {}}
      renderAll={false}
      showBackdrop={true}
      size="auto"
    >
      <div
        className=""
      >
        <Link
          to="/auth/login"
        >
          <Button
            className=""
            size="large"
          >
            <span
              className=""
            >
              Sign in
            </span>
          </Button>
        </Link>
      </div>
      <div
        className=""
      >
        <Link
          to="/auth/registerres"
        >
          <Button
            className=""
            size="large"
          >
            <span
              className=""
            >
              Add your restaurant
            </span>
          </Button>
        </Link>
      </div>
    </Drawer>
  </div>,
]
`;
